R/derive_param_response.R

```diff
@@ -0,0 +1,248 @@
+#' Add a Parameter Indicating If a Subject Had a Response
+#'
+#' Add a new parameter indicating if a response has been observed (AVALC and
+#' AVAL).
+#' If a response has been observed, `AVALC` is set to "Y", `AVAL` to 1 and `ADT`
+#'  is set to the
+#' first date when a response has been observed.
+#' If a response has not been observed, `AVALC` is set to "N", AVAL to 0 and
+#' `ADT` is set NA.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the `subject_keys`and `ADT` are expected.
+#'
+#' @param dataset_adsl Input dataset
+#'
+#'   + The variables specified for `subject_keys` are expected.
+#'   + For each observation of the specified dataset a new observation is added
+#'   to the input dataset.
+#'
+#' @param source_datasets Source dataset(s)
+#'
+#'   A named list of datasets with 1 or 2 elements is expected (e.g. list( adrs
+#'   = adrs, pdds=pdds)).
+#'
+#'   At least `dataset` must be defined as a named list.
+#'
+#'   + If 1 element is defined, it implies that the response data and the data
+#'   used in `pd_source` are in the same dataset `dataset`.
+#'
+#'   + If 2 elements are defined, then 1 dataset contains the response data
+#'   (`dataset`) and the other the data used in `pd_source` (e.g.
+#'   `pd_source$dataset_name`)
+#'
+#'   The variables specified by the `subject_keys` and `ADT` are expected in
+#'   the input `dataset`.
+#'
+#' @param filter_source Source filter
+#'
+#'   All observations in the `dataset` data fulfilling the specified condition
+#'   are selected.
+#'
+#' @param source_pd Sources and conditions defining the end of the assessment
+#' period for the responses.
+#'
+#'   An object of type `date_source` is expected
+#'
+#'   All observations in `dataset` defining the response data fulfilling the
+#'   `filter_source` condition are considered as response if they fall before
+#'   the end of the assessment period as defined by `source_pd`.
+#'
+#'   For example:
+#'   `source_pd = date_source (
+#'     datasets_name="adrs",
```

Comment:
dataset_name instead of datasets

Reply:
also i left a similar comment for filter_pd PR for @bundfussr that i think we should set the PD date once and then input it here as list(pd) so we might not need to even include this example here

Reply:
i would just remove this example block. we'll have it covered in the ADRS vignette